JavaScript SDK Releases — June 2026 (detailed) (#137)
getsentry/sentry-changelog: content/changelog/2026-06-javascript-sdk-releases.md

author: rahulchhabria@sentry.io
---

Releases covered:

| Version | Date | Link |
|---------|------|------|
| 10.62.0 | 2026-06-26 | [Release notes](https://github.com/getsentry/sentry-javascript/releases/tag/10.62.0) |
| 10.61.0 | 2026-06-25 | [Release notes](https://github.com/getsentry/sentry-javascript/releases/tag/10.61.0) |
| 10.60.0 | 2026-06-23 | [Release notes](https://github.com/getsentry/sentry-javascript/releases/tag/10.60.0) |
| 10.59.0 | 2026-06-19 | [Release notes](https://github.com/getsentry/sentry-javascript/releases/tag/10.59.0) |
| 10.58.0 | 2026-06-15 | [Release notes](https://github.com/getsentry/sentry-javascript/releases/tag/10.58.0) |
| 10.57.0 | 2026-06-09 | [Release notes](https://github.com/getsentry/sentry-javascript/releases/tag/10.57.0) |
| 10.56.0 | 2026-06-02 | [Release notes](https://github.com/getsentry/sentry-javascript/releases/tag/10.56.0) |

## TL;DR

> - `streamGenAiSpans` is now enabled by default — `gen_ai` spans stream as v2 envelope items, preventing drops and disabling default message truncation; self-hosted users can opt out with `streamGenAiSpans: false`.

Reviewer: ```suggestion
- `streamGenAiSpans` is now enabled by default — `gen_ai` spans are sent as v2 envelope items, preventing size limit drops and disabling default message truncation; self-hosted users can opt out with `streamGenAiSpans: false`.
```

- Expanded Cloudflare instrumentation: R2 buckets, D1 batch/exec/withSession, SQLite Durable Objects SQL API, and sync KV are all now auto-instrumented.

> - `vercelAiIntegration` adds Vercel AI SDK v7 support; AWS SDK clients ≥ 3.1046.0 are auto-instrumented; Bun and Deno get orchestrion runtime hooks.

Reviewer: ```suggestion
- `vercelAiIntegration` adds Vercel AI SDK v7 support
```

We don't want to highlight these to users.

> - New top-level `Sentry.setAttribute(s)` APIs; Hono transactions named after matched route handlers; `bindScopeToEmitter` for event-emitter-scoped traces.

Reviewer: ```suggestion
- New top-level `Sentry.setAttribute(s)` APIs; Hono transactions named after matched route handlers
```
This is not user-facing.

- React Router v6/v7 navigation tracing fix; PostgresJS duplicate span fix; Next.js redirect classification fix.

## Release notes

### New Features

10.61.0 enables `streamGenAiSpans` by default. `gen_ai` spans are extracted from transactions and sent as v2 envelope items, which means they can no longer be dropped when the transaction payload exceeds size limits, and AI message data is no longer truncated by default. Pass `enableTruncation: true` on the respective AI integration to re-enable truncation. Self-hosted Sentry users should set `streamGenAiSpans: false` until their instance supports streamed spans.

Cloudflare storage instrumentation expanded across four releases: 10.60.0 added R2 bucket auto-instrumentation, 10.61.0 added D1 batch operations, `exec()`, and `withSession()`, as well as SQL API instrumentation for SQLite Durable Objects. 10.59.0 added synchronous KV instrumentation. Together these give complete zero-config coverage for Cloudflare storage primitives.

> 10.62.0 extends `vercelAiIntegration` to support v7 of the Vercel AI SDK (note: not yet available on Cloudflare Workers). In 10.59.0, AWS SDK clients at v3.1046.0 or later are automatically instrumented. Bun gained an orchestrion build plugin in 10.59.0 and Deno a runtime hook, enabling zero-config instrumentation for those runtimes.

Reviewer: ```suggestion
10.62.0 extends `vercelAiIntegration` to support v7 of the Vercel AI SDK (note: not yet available on Cloudflare Workers).
```
We don't want to highlight these to users.

> 10.61.0 adds two new top-level APIs — `Sentry.setAttribute(key, value)` and `Sentry.setAttributes(attributes)` — for setting custom attributes on the currently active span without a direct span reference. Hono transactions are now named after the matched route handler. 10.60.0 introduces `bindScopeToEmitter(emitter)`, which propagates the active Sentry scope to all listeners of a Node.js `EventEmitter`.

Reviewer: ```suggestion
10.61.0 adds two new top-level APIs — `Sentry.setAttribute(key, value)` and `Sentry.setAttributes(attributes)` — for setting custom attributes on the currently active span without a direct span reference. Hono transactions are now named after the matched route handler. 
```
This is not user-facing.
